src/parseTools.js

```diff
@@ -885,11 +885,16 @@ function receiveI64ParamAsI53Unchecked(name) {
   return `var ${name} = convertI32PairToI53(${name}_low, ${name}_high);`;
 }
 
-function sendI64Argument(low, high) {
+// Given the name of a variable containing an unsigned 53-bit integer in a JS
+// Number or BigInt, send it as a 64-bit argument in a call to wasm. This will
+// legalize the argument if necessary (i.e., split it into two i32 arguments,
+// if legalization is used).
+function sendU53ToI64Param(name) {
+  // TODO: In ASSERTIONS mode add range checks here.
   if (WASM_BIGINT) {
-    return 'BigInt(low) | (BigInt(high) << 32n)';
+    return `BigInt(${name})`;
   }
-  return low + ', ' + high;
+  return `(${name} | 0), (${name} / (2**32))`;
```

Comment:
This seems like it should live in `src/library_int53.js`?

Reply:
It's such a tiny code fragment that I think it can stay in the helper function here? Pulling it out would then mean writing deps for it etc.

I do agree that if we add checking then that would be pulled into a library function, as that is no longer trivial.

Reply:
Maybe add a comment saying that this function logically is in the same family as the functions in src/library_int53.js

Reply:
The code in library_int53.js uses expanded out numbers rather than `2**32`.

You could do that here by putting the 2**32 part inside a `${}` ?

Reply:
Wait, why are we expanding them there? It's larger to expand. I think we should `2**32`ify them?

Reply:
Well we should at least be consistent.  I don't really care which form we use.   If you want to use the expanded form in the PR  and followup with change to the runtime multiple everywhere that is also fine be me.